pkg/build/api/v1beta1/types.go

```diff
@@ -223,6 +228,17 @@ type WebHookTrigger struct {
 	Secret string `json:"secret,omitempty" yaml:"secret,omitempty"`
 }
 
+// BuildTriggerImageChangeParams represents the parameters to the ImageChange trigger.
+type ImageChangeTrigger struct {
+	// Image is used to specify the value in the BuildConfig to replace with the 
+	// immutable image id supplied by the ImageRepository when this trigger fires.
+	Image string `json:"image,omitempty" yaml:"image,omitempty"`
+	// ImageRepositoryRef a reference to a Docker image repository to watch for changes.
+	ImageRepositoryRef kapi.ObjectReference `json:"imageRepositoryRef,omitempty" yaml:"imageRepositoryRef,omitempty"`
+	// Tag is the name of an image repository tag to watch for changes.
```

Comment:
If tag isn't optional, we should make it not optional on all APIs.  Or is tag not optional in the other places its used?

----- Original Message -----
> > @@ -223,6 +228,17 @@ type WebHookTrigger struct {
> >  	Secret string `json:"secret,omitempty" yaml:"secret,omitempty"`
> >  }
> >  
> > +// BuildTriggerImageChangeParams represents the parameters to the
> > ImageChange trigger.
> > +type ImageChangeTrigger struct {
> > +	// Image is used to specify the value in the BuildConfig to replace with
> > the
> > +	// immutable image id supplied by the ImageRepository when this trigger
> > fires.
> > +	Image string `json:"image,omitempty" yaml:"image,omitempty"`
> > +	// ImageRepositoryRef a reference to a Docker image repository to watch
> > for changes.
> > +	ImageRepositoryRef kapi.ObjectReference
> > `json:"imageRepositoryRef,omitempty" yaml:"imageRepositoryRef,omitempty"`
> > +	// Tag is the name of an image repository tag to watch for changes.
> 
> I need to update this, none of these fields are going to be optional.
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/557/files#r22596231